superduperdb/backends/mongodb/artifacts.py

```diff
@@ -57,3 +77,76 @@ def disconnect(self):
         """
 
         # TODO: implement me
+
+
+def upload_file(path, file_id, fs):
+    """Upload file to GridFS"""
+    logging.info(f"Uploading file {path} to GridFS with file_id {file_id}")
+    path = Path(path)
+    with open(path, 'rb') as file_to_upload:
+        fs.put(
+            file_to_upload,
+            filename=path.name,
+            metadata={"file_id": file_id, "type": "file"},
+        )
+
+
+def upload_folder(path, file_id, fs, parent_path=""):
+    """Upload folder to GridFS"""
+    path = Path(path)
+    if not parent_path:
+        logging.info(f"Uploading folder {path} to GridFS with file_id {file_id}")
+        parent_path = os.path.basename(path)
+
+    # if the folder is empty, create an empty file
+    if not os.listdir(path):
+        fs.put(
+            b'',
+            filename=os.path.join(parent_path, os.path.basename(path)),
+            metadata={"file_id": file_id, "is_empty_dir": True, 'type': 'dir'},
+        )
+    else:
+        for item in os.listdir(path):
+            item_path = os.path.join(path, item)
+            if os.path.isdir(item_path):
+                upload_folder(item_path, file_id, fs, os.path.join(parent_path, item))
+            else:
+                with open(item_path, 'rb') as file_to_upload:
+                    fs.put(
+                        file_to_upload,
+                        filename=os.path.join(parent_path, item),
+                        metadata={"file_id": file_id, "type": "dir"},
+                    )
+
+
+def download(file_id, fs):
+    """Download file or folder from GridFS and return the path"""
+    temp_dir = tempfile.mkdtemp(prefix=file_id)
+
+    # try to download a file first, if it fails, assume it's a folder
```

Comment:
this comment is not valid anymore right since you have a metadata.type with 'dir' or 'file' which will give you if folder or file.

Reply:
When downloading, we currently only have file_id information and no other information. Although we can also use file_id to judge first, it will increase a search. 

Or save more information upstream, but actually increase the complexity

WDTY?

Reply:
Done